genome_kit/data_manager.py

```diff
@@ -255,3 +257,106 @@ def upload_file(self, filepath: str, filename: str, metadata: Dict[str, str]=Non
     def list_available_genomes(self):
         blobs = self.bucket.list_blobs(match_glob="*.{2bit,cfg}")
         return [blob.name.rpartition(".")[0] for blob in blobs]
+
+class DefaultDataManager(DataManager):
+    """A minimal data manager implementation that retrieves files from a S3 bucket.
+    When using the default bucket, uploads are only supported for GenomeKit maintainers."""
+    def __init__(self, data_dir: str, bucket_name=_S3_BUCKET, require_auth=False):
+        """
+        Args:
+            data_dir: location where local files are cached
+            bucket_name: S3 bucket. By default, use the bucket sponsored by AWS Open Data
+            require_auth: set True for non-public buckets
+        """
+        self._bucket_name = bucket_name
```

Comment:
personal interest: any rationale for using the client API rather than the bucket one?

Reply:
I started with the bucket API but it's missing functionality, like getting object metadata etc. Happy to be corrected if I missed something.

Reply:
maybe add a quick comment inline